### ISSUE TITLE
rockpaper.cpp needs .dat files

### ISSUE THREAD

**Issue by gbowne1** [gbowne1/random_c_cpp_programs#21]
Never was completely sure about the type of files, but when you first run the program it tries to find at least one of the required .dat files.  If they don't exist, they need to be created..

I never decided on what type of file they would be.  

I don't know anything about .dat files.. 

Not sure where to go from here.

**Comment by andrewlod** [comment]
A `.dat` file contains binary data for programs to load. When I tested the program, I wrote `R P S S P R` on `filea.dat` and `S P R M R S` on `fileb.dat` (the `M` is on purpose in order to test invalid characters), and it worked, is that correct? If so, I suppose using `.txt` format, as it is just some text data, would be the best approach.

**Comment by gbowne1** [comment]
At the time I wrote this, .dat seemed to make the most sense, but I am open to other filetypes..  but at this point I never really decided what the file would contain.  I understand the concept of .dat files.  Since it's a binary filetype, do we need to create it in a binary form?

does the program actually create, etc. this file, less should it?

**Comment by gbowne1** [comment]
That being said I have a similar issue with https://github.com/gbowne1/StarshipAscension (which is also a bit of a mess).

**Comment by andrewlod** [comment]
Extension types do not enforce the content of the file, so you can write text data in a `.dat` file (you could even write text on a `.exe` file, or music on a `.png` file). Extensions are more of a suggestion so the OS knows which program to use to open a given file.

This is how you would write and read to binary files, such as `.dat`:
```cpp
#include <iostream>
#include <fstream>

int main(int argc, char const *argv[])
{
    {
        // Write file in binary mode with numbers 1 22 3, delimited by new line
        std::ofstream fileout("file.dat", std::ios::binary);
        fileout << 1 << std::endl << 22 << std::endl << 3;
    }

    {
        // Read file recently created
        std::ifstream filein("file.dat", std::ios::binary);
        while (filein.good()) {
            int number;
            filein >> number;
            std::cout << number << std::endl;
        }
    }
    return 0;
}

```

**Comment by gbowne1** [comment]
I actually just used .dat because in my mind, a large number of the games I have run across used .dat files in some manner, so that made sense.  

I actually like that idea.

**Comment by andrewlod** [comment]
Yeah .dat is widely used among games. It is useful to store more complex data structures and it is slightly cheaper I/O-wise, compared to text files.

**Comment by gbowne1** [comment]
that was my thought as well. I need to do this to StarshipAscension.  though I might want to have separate save game files.  I really need to redo the playing field over there and fix some issues.
